waltz-tools/src/main/java/com/wepay/waltz/tools/client/ClientCli.java

```diff
@@ -521,6 +534,86 @@ protected String getUsage() {
+                .hasArg()
+                .build();
+            cliCfgOption.setRequired(true);
+
+            options.addOption(cliCfgOption);
+        }
+
+        @Override
+        protected void processCmd(CommandLine cmd) throws SubCommandFailedException {
+            ZooKeeperClient zkClient = null;
+            try {
+                String cliConfigPath = cmd.getOptionValue("cli-config-path");
+                CliConfig cliConfig = CliConfig.parseCliConfigFile(cliConfigPath);
+                String zookeeperHostPorts = (String) cliConfig.get(CliConfig.ZOOKEEPER_CONNECT_STRING);
+                String zkRoot = (String) cliConfig.get(CliConfig.CLUSTER_ROOT);
+                int zkSessionTimeout = (int) cliConfig.get(CliConfig.ZOOKEEPER_SESSION_TIMEOUT);
+
+                zkClient = new ZooKeeperClientImpl(zookeeperHostPorts, zkSessionTimeout);
+                ZNode root = new ZNode(zkRoot);
+
+                Set<Endpoint> serverEndpoints = getListOfServerEndpoints(root, zkClient);
+                WaltzClientConfig waltzClientConfig = getWaltzClientConfig(cliConfigPath, false);
+                Map<Endpoint, Map<String, Boolean>> connectivityStatusMap =
+                    checkServerConnections(waltzClientConfig,
+                    serverEndpoints);
+
+                for (Map.Entry<Endpoint, Map<String, Boolean>> connectivityStatus : connectivityStatusMap.entrySet()) {
+                    System.out.println("Connectivity status of " + connectivityStatus.getKey() + " is: "
+                        + ((connectivityStatus.getValue() == null) ? "UNREACHABLE"
+                        : connectivityStatus.getValue().toString()));
+                }
+            } catch (Exception e) {
+                throw new SubCommandFailedException(String.format("Failed to check topology of the cluster. %n%s",
+                    e.getMessage()));
+            } finally {
+                if (zkClient != null) {
+                    zkClient.close();
+                }
+            }
+        }
+
+        public Set<Endpoint> getListOfServerEndpoints(ZNode root, ZooKeeperClient zkClient) throws Exception {
+            ClusterManager clusterManager = new ClusterManagerImpl(zkClient, root, partitionAssignmentPolicy);
+            Set<ServerDescriptor> serverDescriptors = clusterManager.serverDescriptors();
+            Set<Endpoint> serverEndpoints = new HashSet<>();
+
+            for (ServerDescriptor serverDescriptor: serverDescriptors) {
+                serverEndpoints.add(serverDescriptor.endpoint);
+            }
+            return serverEndpoints;
+        }
+
+        private Map<Endpoint, Map<String, Boolean>> checkServerConnections(WaltzClientConfig config,
+                                                                           Set<Endpoint> serverEndpoints) throws Exception {
+            DummyTxnCallbacks callbacks = new DummyTxnCallbacks();
+            WaltzClient client = new WaltzClient(callbacks, config);
```

Comment:
Creating a `WaltzClient` for every invocation is heavy. Will you consider using `InternalRpcClient` here?
It is very strange for `WaltzClient` to have `checkServerConnections(Set<Endpoint> serverEndpoints)`.